nemo_aligner/data/nlp/builders.py

```diff
@@ -321,29 +321,39 @@ def build_dataloader(
     pad_samples_to_global_batch_size=False,
     collate_fn=None,
     load_gbs=True,
+    use_random_sampler=True,
 ):
     """Buld dataloader given an input dataset."""
 
+    def create_batch_sampler(cls, **kwargs):
+        """Helper function to create a batch sampler."""
+        return cls(**kwargs)
+
     logging.info(f"Building dataloader with consumed samples: {consumed_samples}")
+    # Common parameters for batch sampler creation
+    common_params = {
+        "total_samples": len(dataset),
+        "consumed_samples": consumed_samples,
+        "micro_batch_size": mbs,
+        "data_parallel_rank": parallel_state.get_data_parallel_rank(),
+        "data_parallel_size": parallel_state.get_data_parallel_world_size(),
+        "drop_last": drop_last,
+        "global_batch_size": gbs,
+        "pad_samples_to_global_batch_size": pad_samples_to_global_batch_size,
+    }
+
```

Comment:
can you add a if `use_random_sampler` and then put the seed into the common_params.

this way we don't need the `create_batch_sampler` function

Reply:
Sure. Fixed.